contracts/GameJamProxy.sol

```diff
@@ -0,0 +1,70 @@
+pragma solidity 0.5.12;
+
+import "openzeppelin-solidity/contracts/access/Roles.sol";
+import "./interfaces/IRegistry.sol";
+import "./GameJamCommon.sol";
+
+
+contract GameJamProxy is GameJamCommon {
+  // Define a ContractRegistry to be setup in the Constructor
+  IRegistry registry;
+
+  // Define address the Proxied contract
+  address public gameJamContractAddress;
+
+  constructor(
+    address _registry,
+    address _gameJamContractAddress
+  )
+    public
+    payable
+  {
+    require(_registry != address(0));
+
+    gameJamContractAddress = _gameJamContractAddress;
+    registry = IRegistry(_registry);
+  }
+
+  function()
+    external
+    payable
+  {
+    assembly {
+      // Load gameJam address from first storage pointer
+      let _gameJam := sload(gameJamContractAddress_slot)
+
+      // calldatacopy(t, f, s)
+      calldatacopy(
+        0x0, // t = mem position to
+        0x0, // f = mem position from
+        calldatasize // s = size bytes
+      )
+
+      // delegatecall(g, a, in, insize, out, outsize) => returns "0" on error, or "1" on success
+      let result := delegatecall(
+        gas, // g = gas
+        _gameJam, // a = address
+        0x0, // in = mem in  mem[in..(in+insize)
+        calldatasize, // insize = mem insize  mem[in..(in+insize)
+        0x0, // out = mem out  mem[out..(out+outsize)
+        0 // outsize = mem outsize  mem[out..(out+outsize)
+      )
+
+      if iszero(result) {
+        revert(0, 0)
+      }
+
+      // returndatacopy(t, f, s)
+      returndatacopy(
+        0x0, // t = mem position to
+        0x0,  // f = mem position from
+        returndatasize // s = size bytes
```

Comment:
What is the purpose of returndatacopy here?

Reply:
> Copies data from the return data buffer to memory

It is an OPCODE for the EVM, have a look here => https://github.com/crytic/evm-opcodes

and here => https://github.com/ethereum/EIPs/blob/master/EIPS/eip-211.md

These would do a better job explaining that I can.

Also tbh this code is mostly copy/pasta, so I'm actually not 100% sure myself 😂